apps/crn-server/src/data-providers/users.ts

```diff
@@ -0,0 +1,338 @@
+    }
+  };
+  const fetch = async (options: FetchUsersOptions) => {
+    const queryFilter = generateFetchQueryFilter(options);
+    const { take = 8, skip = 0 } = options;
+    const users = await queryForUsers(
+      squidexGraphlClient,
+      queryFilter,
+      take,
+      skip,
+    );
+    return users.map(parseGraphQLUserToDataObject);
+  };
+  const fetchByCode = async (code: string) => {
+    const filter = `data/connections/iv/code eq '${code}'`;
+    const users = await queryForUsers(squidexGraphlClient, filter, 1, 0);
+    return users.map(parseGraphQLUserToDataObject);
+  };
+  const updateAvatar = async (
+    id: string,
+    avatar: Buffer,
+    contentType: string,
+  ) => {
+    const assetId = await uploadAvatar(
+      userSquidexRestClient,
+      id,
+      avatar,
+      contentType,
+    );
+
+    await userSquidexRestClient.patch(id, { avatar: { iv: [assetId] } });
+  };
+
+  const connectByCode = async (welcomeCode: string, userId: string) => {
+    const user = await queryByCode(welcomeCode, userSquidexRestClient.client);
+
+    if (!user) {
+      return null;
+    }
+
+    if (user.data.connections.iv?.find(({ code }) => code === userId)) {
+      return parseUserToDataObject(user);
+    }
+
+    const connections = (user.data.connections.iv || []).concat([
+      { code: userId },
+    ]);
+
+    const res = await userSquidexRestClient.patch(user.id, {
+      email: { iv: user.data.email.iv },
+      connections: { iv: connections },
+    });
+
+    return parseUserToDataObject(res);
+  };
+  const syncOrcidProfile = async (
+    id: string,
+    cachedUser: RestUser | undefined = undefined,
+  ) => {
+    let fetchedUser;
+    if (!cachedUser) {
+      fetchedUser = await userSquidexRestClient.fetchById(id);
+    }
+
+    const user = cachedUser || (fetchedUser as RestUser);
+
+    const [error, res] = await Intercept(
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      fetchOrcidProfile(user!.data.orcid!.iv),
+    );
+
+    const updateToUser: Partial<RestUser['data']> = {
+      email: { iv: user.data.email.iv },
+      orcidLastSyncDate: { iv: new Date().toISOString() },
+    };
+
+    if (!error) {
+      const { lastModifiedDate, works } = transformOrcidWorks(res);
+      updateToUser.orcidLastModifiedDate = { iv: lastModifiedDate };
+      updateToUser.orcidWorks = { iv: works.slice(0, 10) };
+    }
+
+    const updatedUser = await userSquidexRestClient.patch(
+      user.id,
+      updateToUser,
+    );
+    return parseUserToDataObject(updatedUser);
+  };
+  return {
+    fetchById,
+    update,
+    fetch,
+    fetchByCode,
+    updateAvatar,
+    connectByCode,
+    syncOrcidProfile,
+  };
+}
+const shouldDoFullUpdate = (userToUpdate: UserPatchDataObject) =>
+  userToUpdate.teams?.length ||
+  Object.values(userToUpdate).some(
+    (value) => value.trim && value.trim() === '',
+  );
+
+async function queryForUsers(
```

Comment:
shouldnt we just use consts with arrow functions?

Reply:
done